modules/cli/src/main/scala/scala/cli/commands/Run.scala

```diff
@@ -75,10 +99,18 @@ object Run extends ScalaCommand[RunOptions] {
         partial = None,
         postAction = () => WatchUtil.printWatchMessage()
       ) { res =>
+        for ((process, onExitProcess) <- processOpt) {
+          onExitProcess.cancel(true)
+          ProcUtil.interruptProcess(process, logger)
```

Comment:
It might not be enough to send a SIGINT… If the process doesn't exit, we shouldn't proceed with running a new one straightaway. Maybe just before the call to `maybeRun` below, we could check again if the process is still alive, and send it a SIGKILL with `Process.destroyForcibly` (and only then, warn users if ever it's still alive and start a new one anyway).